From dc2e1e24e5a69face7d154fea6d3ecbee6c90e45 Mon Sep 17 00:00:00 2001
From: Arif Alam <arif.alam@netexperience.com>
Date: Wed, 28 Apr 2021 19:29:23 -0400
Subject: [PATCH] netifd: Add WPA3 Enterprise modes

Add configuration options for:
- WPA3 Enterprise Only mode
- WPA3 Enterprise Transition mode

Signed-off-by: Arif Alam <arif.alam@netexperience.com>
---
 .../patches/0105-add-wpa3-enterprise-modes.patch  | 15 +++++++++++++++
 1 file changed, 15 insertions(+)
 create mode 100644 package/network/config/netifd/patches/0105-add-wpa3-enterprise-modes.patch

diff --git a/package/network/config/netifd/patches/0105-add-wpa3-enterprise-modes.patch b/package/network/config/netifd/patches/0105-add-wpa3-enterprise-modes.patch
new file mode 100644
index 0000000000..9018365807
--- /dev/null
+++ b/package/network/config/netifd/patches/0105-add-wpa3-enterprise-modes.patch
@@ -0,0 +1,15 @@
+--- a/scripts/netifd-wireless.sh
++++ b/scripts/netifd-wireless.sh
+@@ -244,8 +244,11 @@ wireless_vif_parse_encryption() {
+ 		owe*)
+ 			auth_type=owe
+ 		;;
++		wpa3-only*)
++			auth_type=eap-only
++		;;
+ 		wpa3-mixed*)
+-			auth_type=eap-eap192
++			auth_type=eap-transition
+ 		;;
+ 		wpa3*)
+ 			auth_type=eap192
-- 
2.25.1

